src/plugins/plotlyPlot/components/PlotlyViewLayout.vue

```diff
@@ -0,0 +1,221 @@
+<template>
+<div class="l-view-section"></div>
+</template>
+
+<script>
+import Plotly from 'plotly.js-dist';
+import moment from 'moment'
+
+export default {
+    inject: ['openmct', 'domainObject'],
+    data: function () {
+
+        return {
+            telemetryObjects: [],
+            bounds: this.openmct.time.bounds(),
+            timeRange:  0,
+            plotData: {},
+            subscriptions: {}
+        }
+    },
+    mounted() {
+        this.plotElement = document.querySelector('.l-view-section');
+
+        this.composition = this.openmct.composition.get(this.domainObject);
+        this.composition.on('add', this.addTelemetry);
+        this.composition.on('remove', this.removeTelemetry);
+        this.composition.load();
+
+        this.openmct.time.on('bounds', this.refreshData);
+        this.openmct.time.on('clock', this.changeClock);
+    },
+    destroyed() {
+        this.unsubscribe();
+    },
+    methods: {
+        changeClock() {
+            if (this.openmct.time.clock()) {
+                Plotly.purge(this.plotElement);
+                this.telemetryObjects.forEach((telemetryObject, index) => {
+                    this.subscribeTo(telemetryObject, index);
+                });
+            }
+        },
+        addTelemetry(telemetryObject) {
+            this.telemetryObjects.push(telemetryObject);
+            const index = this.telemetryObjects.findIndex(obj => obj === telemetryObject);
+            this.requestHistory(telemetryObject, index, true);
+            this.subscribeTo(telemetryObject, index);
+        },
+        subscribeTo(telemetryObject, index) {
+            let keyString = this.openmct.objects.makeKeyString(telemetryObject.identifier);
+            this.subscriptions[keyString] = this.openmct.telemetry.subscribe(telemetryObject, (datum) => {
+                //Check that telemetry object has not been removed since telemetry was requested.
```

Comment:
i think if you want to check if the telemetry has been removed since the request you'd need to put this check outside of the subscription. I'm not sure its necessary at all though.